service/topology_coordinator.cc

```diff
@@ -850,6 +853,90 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
+                    const size_t live_replicas = std::count_if(replica_endpoints.begin(), replica_endpoints.end(),
+                                                                [&live_nodes] (const gms::inet_address& addr) {
+                                                                    return live_nodes.contains(addr);
+                                                                });
+                    if (live_replicas == replica_endpoints.size()) {
+                        try {
+                            // Send the RPC to all replicas
+                            const netw::messaging_service::clock_type::time_point timeout_deadline = lowres_clock::now() + timeout;
+                            const service::frozen_topology_guard frozen_guard { _topo_sm._topology.session };
+                            co_await coroutine::parallel_for_each(replica_endpoints, [&] (const gms::inet_address& ep) -> future<> {
+                                netw::messaging_service::msg_addr addr(ep);
+                                co_await ser::storage_proxy_rpc_verbs::send_truncate(&_messaging, addr, timeout_deadline, ks_name, cf_name, frozen_guard);
+                            });
+                        } catch (std::exception& e) {
+                            error = e.what();
+                        }
+                    } else {
+                        error = ::format("Cannot perform TRUNCATE on column family {}.{} because some hosts are down", ks_name, cf_name);
+                    }
+                } else {
+                    error = ::format("Column family with UUID {} does not exist.", table_id);
+                }
+
+                if (error) {
+                    rtlogger.warn("Error while TRUNCATE TABLE: {}", error.value());
+                }
+
+                std::vector<canonical_mutation> updates;
+                updates.push_back(topology_mutation_builder(guard.write_timestamp())
+                                    .del_global_topology_request()
+                                    .del_global_topology_request_id()
+                                    .del_session()
+                                    .build());
+                updates.push_back(topology_request_tracking_mutation_builder(global_request_id)
+                                    .done(error)
+                                    .build());
+
+                sstring reason = "Truncate has completed";
+                topology_change change{std::move(updates)};
+                group0_command g0_cmd = _group0.client().prepare_command(std::move(change), guard, reason);
+                try {
+                    co_await _group0.client().add_entry(std::move(g0_cmd), std::move(guard), _as);
+                    break;
+                } catch (group0_concurrent_modification&) {
+                    rtlogger.info("handle_global_request(): concurrent modification, retrying");
```

Comment:
We shouldn't retry the whole truncate operation on group0_concurrent_modification. This is potential liveness problem.